sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlSuite.scala

```diff
@@ -2685,4 +2684,40 @@ class XmlSuite extends QueryTest with SharedSparkSession {
         Row(1.0E38D, BigDecimal("92233720368547758070")) :: Nil
     )
   }
+
+  test("return partial results for bad records") {
+    withTempDir { dir =>
+      val xmlBadRecord1 =
```

Comment:
Add a scenario with nested struct object with attributes and valueTag:
Correct data:
`<ROW><struct attr=3.0>4.0<b>5.0</b><struct></ROW>`

XML data with mismatches:
```
<ROW><struct attr=mismatch1>4.0<b>5.0</b><struct></ROW>
<ROW><struct attr=3.0>mismatch2<b>5.0</b><struct></ROW>
<ROW><struct attr=3.0>4.0<b>mismatch3</b><struct></ROW>
```